### ISSUE TITLE
Real-time Display of Preliminary HCHO Mixing Ratios

### ISSUE THREAD

**Issue by jdshutter** [nthallen/keutsch-hcho#4]
This would require using a (1) pre-determined cal factor and (2) power meter calibration after having found the difference counts between the online and offline positions.

**Comment by Mitchellpkt** [comment]
@jdshutter For preliminary data, I wonder if you could bypass need (2) for power meter calibration, by normalizing to UDT voltage instead of laser power? As long as nobody futzes with the alignment, I'd expect a reasonably-stable "rough-draft cal factor" relating (Diff Counts)/(seconds * mV) to ppb HCHO.

**Comment by nthallen** [comment]
Remind me: what is UDT voltage?

**Comment by jdshutter** [comment]
UDT voltage is just the voltage from the power meter. Mitchell is right that we can simply use it as a rough cal factor. I've been normally converting UDT voltage to mW in order to compare everything to Tom's ISAF instrument.

**Comment by nthallen** [comment]
OK, sure, but you still need a calibration to get the cal factor relative to UDT voltage. As you say, if the alignment changes, all bets are off, whether we're talking volts or mW.

**Comment by Mitchellpkt** [comment]
Whoops, I think "UDT" is a brand/model identifier. Sorry, I meant to indicate normalization to the instrument's permanent beam power meter (measuring real-time laser power into &/or out of the cell)

The only difference is practical considerations that arose in the field with the old HCHO instruments. Since no disassembly is necessary to acquire the "rough draft" cal factor normalized to voltage, it was convenient to measure between flights in broad daylight while bolted down to the aircraft. However, the extra step to calibrate the power meter (laser wattage <--> power meter voltage) required taking apart the optical box in a dark room. More of a twice-per-campaign task, as opposed to re-measuring the "rough draft" cal factor routinely.

True, both cal factors change after any alignment changes -- with our previous instrument configurations, it was just WAY easier to update the voltage-normalized cal factor on-the-fly.

**Comment by nthallen** [comment]
Does it make sense to think of (Diff Counts)/(seconds * mV) to ppb HCHO as the primary calibration, with the mV to mW calibration being secondary? Referring to the former as only rough cut doesn't really make sense to me in this context.

**Comment by Mitchellpkt** [comment]
Yes & agree. I think of the voltage-normalized value as the "raw calibration factor," since the per-mW version is essentially just calculated by unit conversion with the measured mW/mV ratio.

**Comment by jdshutter** [comment]
The real-time display is working well, Norton! Would it be possible to add this as a variable that could be plotted?
